src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IForEachLoopStatement.cs

```diff
@@ -3805,16 +3806,18 @@ static async System.Threading.Tasks.Task Main(System.Collections.Generic.IAsyncE
     }
 }
 ";
-            // https://github.com/dotnet/roslyn/issues/30362 how do we flag `await`?
             string expectedOperationTree = @"
-IForEachLoopOperation (LoopKind.ForEach, Continue Label Id: 0, Exit Label Id: 1) (OperationKind.Loop, Type: null) (Syntax: 'await forea ... }')
+IForEachLoopOperation (LoopKind.ForEach, IsAsynchronous, Continue Label Id: 0, Exit Label Id: 1) (OperationKind.Loop, Type: null) (Syntax: 'await forea ... }')
   Locals: Local_1: System.String value
   LoopControlVariable: 
     IVariableDeclaratorOperation (Symbol: System.String value) (OperationKind.VariableDeclarator, Type: null) (Syntax: 'string')
       Initializer: 
         null
   Collection: 
-    IParameterReferenceOperation: pets (OperationKind.ParameterReference, Type: System.Collections.Generic.IAsyncEnumerable<System.String>) (Syntax: 'pets')
+    IConversionOperation (TryCast: False, Unchecked) (OperationKind.Conversion, Type: System.Collections.Generic.IAsyncEnumerable<System.String>, IsImplicit) (Syntax: 'pets')
```

Comment:
Do you know why this conversion is introduced?

Reply:
Not sure. The logic in BindForEachPartsWorker always wraps the collection expression with a BoundConversion (even when the conversion is Identity).
The conversion is there in other IOperation tests for regular foreach loops (`ForEachFlow_06`) so this seems correct.

Let me know if this bugs you. I can investigate further.

---
In reply to: [316309905](https://github.com/dotnet/roslyn/pull/37963#discussion_r316309905) [](ancestors = 316309905)